include/cantera/kinetics/ReactionFactory.h

```diff
@@ -0,0 +1,123 @@
+/**
+ *  @file ReactionFactory.h
+ *  Factory class for reaction functions. Used by classes
+ *  that implement kinetics
+ *  (see \ref reactionGroup and class \link Cantera::Reaction Reaction\endlink).
+ */
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#ifndef CT_NEWREACTION_H
+#define CT_NEWREACTION_H
+
+#include "cantera/base/FactoryBase.h"
+#include "cantera/kinetics/Reaction.h"
+
+namespace Cantera
+{
+
+/**
+ * Factory class to construct reaction function calculators.
+ * The reaction factory is accessed through static method factory:
+ *
+ * @code
+ * Reaction* f = ReactionFactory::factory()->newReaction(type, c)
+ * @endcode
+ *
+ * @ingroup reactionGroup
+ */
+class ReactionFactory : public Factory<Reaction>
```

Comment:
I admit that there may be opportunity to streamline further, but while XML is still around, I'm treading carefully. This doesn't mean that this shouldn't be taken care of eventually.

Reply:
I thought about it some more and don't see a clean way to handle this at the moment (XML creates some wrinkles). As the factory approach needs to work for both XML and AnyMap, I decided to add a `@todo` in the header rather than solving this right now.